calico_cni_k8s_test.go

```diff
@@ -0,0 +1,174 @@
+package main_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"os"
+	"time"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"net"
+
+	"syscall"
+
+	"github.com/containernetworking/cni/pkg/ns"
+	"github.com/containernetworking/cni/pkg/types"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gexec"
+	. "github.com/projectcalico/calico-cni/test_utils"
+	"github.com/projectcalico/libcalico-go/lib/api"
+	"github.com/projectcalico/libcalico-go/lib/backend/k8s"
+	cnet "github.com/projectcalico/libcalico-go/lib/net"
+	"github.com/vishvananda/netlink"
+)
+
+func init() {
+	// Create a random seed
+	rand.Seed(time.Now().UTC().UnixNano())
+}
+
+var _ = Describe("CalicoCni", func() {
+	hostname, _ := os.Hostname()
+	BeforeEach(func() {
+		WipeK8sPods()
+		WipeEtcd()
+	})
+
+	Describe("Run Calico CNI plugin in K8s mode", func() {
+		Context("using host-local IPAM", func() {
+
+			//TODO - set the netconfig
+			netconf := fmt.Sprintf(`
+			{
+			  "name": "net1",
+			  "type": "calico",
+			  "etcd_endpoints": "http://%s:2379",
+			  "ipam": {
+			    "type": "host-local",
+			    "subnet": "10.0.0.0/8"
+			  },
+				"kubernetes": {
+					        "k8s_api_root": "http://127.0.0.1:8080"
+									    
+				},
+				"policy": {"type": "k8s"},
+	"log_level":"info"
+			}`, os.Getenv("ETCD_IP"))
+
+			It("successfully networks the namespace", func() {
+				config, err := clientcmd.DefaultClientConfig.ClientConfig()
+				if err != nil {
+					panic(err)
+				}
+				clientset, err := kubernetes.NewForConfig(config)
+
+				if err != nil {
+					panic(err)
+				}
+				name := fmt.Sprintf("run%d", rand.Uint32())
+				interfaceName := k8s.VethNameForWorkload(fmt.Sprintf("test.%s", name))
+				_, err = clientset.Pods("test").Create(&v1.Pod{
```

Comment:
nit:  Given we rely on the delete code deleting the entries in this namespace perhaps nicer to have this namespace as a const.